src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs

```diff
@@ -0,0 +1,51 @@
+﻿using HE.InvestmentLoans.BusinessLogic.User.Repositories;
+using HE.InvestmentLoans.Common.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace HE.InvestmentLoans.Common.Authorization;
+
+public class LoanUserContext : ILoanUserContext
+{
+    private readonly IUserContext _userContext;
+
+    private readonly ILoanUserRepository _loanUserRepository;
+
+    private Guid? _accountId = null;
+
+    public LoanUserContext(IUserContext userContext, ILoanUserRepository loanUserRepository)
+    {
+        _userContext = userContext;
+        _loanUserRepository = loanUserRepository;
+    }
+
+    public string UserGlobalId => _userContext.UserGlobalId;
+
+    public string? Email => _userContext.Email;
+
+    public IList<string> Roles { get; private set; }
```

Comment:
I think this collection can still be modified from outside the object, for example new roles can be added to collection even though setter is private. We can avoid that using IReadonlyCollection.
```suggestion
    private List<string> _roles;
    public IReadonlyCollection<string> Roles { get=> _roles.AsReadonly(); }
```

Reply:
Good point. For now Roles property is not used. I will fix this in the next PR.